packages/core/src/fhir-to-cda/cda-templates/components/immunizations.ts

```diff
@@ -0,0 +1,200 @@
+
+  return [
+    {
+      tr: {
+        _ID: referenceId,
+        ["td"]: [
+          {
+            _ID: `${referenceId}-material`,
+            "#text": immunization.resource.vaccineCode
+              ? getDisplaysFromCodeableConcepts([immunization.resource.vaccineCode])
+              : NOT_SPECIFIED,
+          },
+          {
+            "#text": mapImmunizationStatusCode(immunization.resource.status) ?? NOT_SPECIFIED,
+          },
+          {
+            "#text": locationInfo ?? immunization.locationName ?? NOT_SPECIFIED,
+          },
+          {
+            "#text":
+              formatDateToHumanReadableFormat(immunization.resource.occurrenceDateTime) ??
+              NOT_SPECIFIED,
+          },
+        ],
+      },
+    },
+  ];
+}
+
+function getLocationInformation(location: Location | undefined): string | undefined {
+  if (!location) return undefined;
+  return `${location.name} - ${buildAddressText(location.address)}`;
+}
+
+function mapImmunizationStatusCode(status: string | undefined): string | undefined {
+  if (!status) return undefined;
+  switch (status) {
+    case "completed":
+      return "completed";
+    case "entered-in-error":
+      return "nullified";
+    case "not-done":
+      return "new";
+    default:
+      return "completed";
+  }
+}
+
+function createEntryFromEncounter(immunization: AugmentedImmunization, referenceId: string) {
+  return {
+    substanceAdministration: {
+      _classCode: "SBADM",
+      _moodCode: "EVN",
+      _negationInd: "false",
+      templateId: buildInstanceIdentifier({
+        root: immunization.typeOid,
+        extension: extensionValue2015,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: immunization.resource.id,
+      }),
+      code: buildCodeCe({
+        code: "IMMUNIZ",
+        codeSystem: hl7actCode,
+        codeSystemName: "ActCode",
+      }),
+      text: buildSimpleReference(referenceId),
+      statusCode: {
+        _code: mapImmunizationStatusCode(immunization.resource.status),
+      },
+      effectiveTime: withoutNullFlavorObject(
+        formatDateToCdaTimestamp(immunization.resource.occurrenceDateTime),
+        "_value"
+      ),
+      consumable: buildConsumable(immunization.resource, referenceId),
+      performer: buildPerformerFromLocation(immunization.location),
+    },
+  };
+}
+
+function buildConsumable(immunization: Immunization, referenceId: string) {
+  return {
+    _typeCode: "CSM",
+    manufacturedProduct: {
+      _classCode: "MANU",
+      templateId: buildInstanceIdentifier({
+        root: oids.immunizationMedicationInformation,
+        extension: extensionValue2015,
+      }),
+      manufacturedMaterial: {
+        code: buildCodeCvFromCodeableConcept(immunization.vaccineCode, `${referenceId}-material`),
+      },
+    },
+  };
+}
+
+function buildSimpleReference(referenceId: string): CdaSimpleReference {
+  return {
```

Comment:
isnt this the same function as `buildOriginalTextReference`?

On a separate note, why are we making a reference to the unstructured section in the text and in the codeableConcept sections?

Reply:
Good eye! Yes, it's the same function. Will take this new one out. 

I see this sorta thing a lot in CDAs, where the structured part is referring to the text/html tables. And this is what we do a lot around the FHIR-to-CDA converter. 

Not sure what you mean by `...and in the codeableConcept sections`

Reply:
ok sounds good.